### PR TITLE
Add lint for useless transmutes

### LINKED ISSUE #441: Lint on useless transmutes
https://github.com/rust-lang/rust/issues/11400

We should lint on transmutes between the same type, e.g. `transmute::<u8,u8>(...)`

The easiest thing to do here is to find `ExprCall`s which resolve to `transmute`, and checking if the `expr_ty` of the argument matches that of the whole call expression.

### PR DESCRIPTION
**Pull request by apasel422**
Closes #441.

### CODE REVIEW

**Comment by Manishearth** [comment]
Run `util/update_lints.py` to update the readme

**Comment by apasel422** [comment]
Updated, thanks!

File: tests/compile-fail/transmute.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+extern crate core;
+
+use std::mem::transmute as my_transmute;
+use std::vec::Vec as MyVec;
+
+fn my_vec() -> MyVec<i32> {
+    vec![]
+}
+
```
Comment (Manishearth) [inline tests/compile-fail/transmute.rs]: Could you add some tests on generics? One test which tests for `transmute<T,T>` (which might trigger the lint), and one for `transmute<T,U>` (which should not). It's okay if the lint doesn't warn on `<T,T>` -- in that case leave a comment on the test to that effect.

Comment (apasel422) [inline tests/compile-fail/transmute.rs]: It's not possible to generically transmute from `T` to `U`:
```
error: cannot transmute to or from a type that contains unsubstituted type parameters [E0139]
```
but I'll add a test from `&T` to `&U`.

Comment (Manishearth) [inline tests/compile-fail/transmute.rs]: Sorry, yeah, I meant something of that form.

Comment (apasel422) [inline tests/compile-fail/transmute.rs]: Done.

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**